### PR TITLE
Add Connection String Execute Step for SQL Bindings

### LINKED ISSUE #19203: SQL Bindings connection string may not be added to local.settings.json
SQL Bindings 0.0.1

**Scenario: User has no folder open and right-clicks table to use command** 

1. Warning to Create Azure Function Project (next steps) or Learn More (aka.ms/sqlbindings) 
2. Browse folder to use for azure function project 
3. Select Type of Binding (input or output) 
4. Function Name - (retrieved from the schema and table name – sanitized and removed brackets) 
5. Provide namespace 
6. Creates azure function with SQL binding 
7. Ask user to include password for connection string in local.settings.json 
8. Add a connection string to local.settings.json 

Step 6 shows this prompt: 
Screen Shot 2022-04-22 at 1 03 10 PM

In Step 6 if a user chooses `open in current window` or `open in a new window`.
VSCode is restarted (with the selected folder).
The extension host will be restarted as well and so the remaining logic after `createFunction API` will not be executed.

### PR DESCRIPTION
**Pull request by VasuBhog**
This PR fixes #19203.
PR in https://github.com/microsoft/vscode-azurefunctions/pull/3150.

The issue was the when a user is creating a new azure function project with the command `Create Azure Function with SQL binding` they would eventually be prompted to open the project:
image

If a user chooses `Open in current window` or `Open in new window` then the logic to add the connectionString to the local.settings.json would not be executed due to the vscode window reloading and therefore the extension host restarting. 

I worked with vscode-azurefunctions team to find a solution that would benefit users generically for azure functions. Allowing for an execute step to execute prior to the OpenFolderStep in order for us to execute the logic with the newly create azure function project that is handled in the createFunction API step. 

This will greatly aid in the user's experience for creating new azure functions with sql bindings as well as aid in future cases where we would like to execute something prior to the openFolderStep.

### CODE REVIEW

**Review by barbaravaldez — APPROVED** [review]

**Review by Charles-Gagnon — APPROVED** [review]
It looks good but unfortunately we can't take this in until they publish a new version of the extension with these changes. Keep an eye on that and merge this in once that comes out and you've verified it works as expected.

File: extensions/sql-bindings/src/services/azureFunctionsService.ts [hunk 1]
```diff
@@ -238,30 +241,18 @@ export async function createAzureFunction(node?: ITreeNodeInfo): Promise<void> {
 				...(selectedBindingType === BindingType.output && { table: objectName })
 			},
 			folderPath: projectFolder,
-			suppressCreateProjectPrompt: true
+			suppressCreateProjectPrompt: true,
+			...(isCreateNewProject && { executeStep: connectionStringExecuteStep })
```
Comment (Charles-Gagnon) [inline extensions/sql-bindings/src/services/azureFunctionsService.ts:245]: So one thing you may want to do is update this to actually take in an array of items (or even better either single or array). That way if we have more steps we want to add later then we don't have to get another version of the extension out.

Comment (VasuBhog) [reply extensions/sql-bindings/src/services/azureFunctionsService.ts:245]: I can look into adding this for future cases as well.

**Comment by VasuBhog** [comment]
Merged this in so that we can have bug bash on latest version of SQL bindings (0.1.0) and follow up issues can get resolved accordingly. In order to use sql-bindings must use insiders package of vscode-azurefunctions.

**Pull request merged**